src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingAssessmentResource.java

```diff
@@ -168,18 +164,9 @@ else if (!newResult.getFeedbacks().isEmpty() && newResult.getFeedbacks().stream(
             throw new BadRequestAlertException("In case feedback is present, a feedback must contain points.", ENTITY_NAME, "feedbackCreditsNull");
         }
 
-        ProgrammingSubmission submission;
-        if (latestExistingResult.isEmpty()) {
-            // Create manual submission with last commit hash and current time stamp.
-            submission = programmingSubmissionService.createSubmissionWithLastCommitHashForParticipation((ProgrammingExerciseStudentParticipation) participation,
-                    SubmissionType.MANUAL);
-            newResult.setSubmission(submission);
-        }
-        else {
-            submission = programmingSubmissionService.findByIdWithEagerResultAndFeedback(latestExistingResult.get().getSubmission().getId());
-            newResult.setSubmission(submission);
-        }
-
+        // make sure that the submission cannot be manipulated on the client side
+        ProgrammingSubmission submission = programmingSubmissionService.findByIdWithEagerResultAndFeedback(latestExistingResult.getSubmission().getId());
```

Comment:
Is this call necessary? latestExistingResult should already have an safe submission fetched from the DB.